pvlib/spectrum.py

```diff
@@ -0,0 +1,363 @@
+        aerosol_transmittance,
+        vapor_transmittance,
+        ozone_transmittance,
+        mixed_transmittance,
+        aerosol_scattering,
+        aerosol_absorption,
+    )
+
+
+def spectrl2(apparent_zenith, aoi, surface_tilt, ground_albedo,
+             surface_pressure, relative_airmass, precipitable_water, ozone,
+             aerosol_turbidity_500nm, dayofyear=None,
+             scattering_albedo_400nm=0.945, alpha=1.14,
+             wavelength_variation_factor=0.095, aerosol_asymmetry_factor=0.65):
+    """
+    Estimate spectral irradiance using the Bird Simple Spectral Model
+    (SPECTRL2).
+
+    The Bird Simple Spectral Model [1]_ produces terrestrial spectra between
+    300 and 4000 nm with a resolution of approximately 10 nm. Direct and
+    diffuse spectral irradiance are modeled for horizontal and tilted surfaces
+    under cloudless skies. SPECTRL2 models radiative transmission, absorption,
+    and scattering due to atmospheric aerosol, water vapor, and ozone content.
+
+    Parameters
+    ----------
+    apparent_zenith : numeric
+        Solar zenith angle [degrees]
+    aoi : numeric
+        Angle of incidence of the solar vector on the panel [degrees]
+    surface_tilt : numeric
+        Panel tilt from horizontal [degrees]
+    ground_albedo : numeric
+        Albedo [0-1] of the ground surface. Can be provided as a scalar value
+        if albedo is not spectrally-dependent, or as a 122xN matrix where
+        the first dimension spans the wavelength range and the second spans
+        the number of simulations. [unitless]
+    surface_pressure : numeric
+        Surface pressure [Pa]
+    relative_airmass : numeric
+        Relative airmass. The airmass model used in [1]_ is the `'kasten1966'`
+        model, while a later implementation by NREL uses the
+        `'kastenyoung1989'` model. [unitless]
+    precipitable_water : numeric
+        Atmospheric water vapor content [cm]
+    ozone : numeric
+        Atmospheric ozone content [atm-cm]
+    aerosol_turbidity_500nm : numeric
+        Aerosol turbidity at 500 nm [unitless]
+    dayofyear : numeric, optional
+        The day of year [1-365].  Must be provided if ``apparent_zenith`` is
+        not a pandas Series.
+    scattering_albedo_400nm : numeric, default 0.945
+        Aerosol single scattering albedo at 400nm. The default value of 0.945
+        is suggested in [1]_ for a rural aerosol model. [unitless]
+    alpha : numeric, default 1.14
+        Angstrom turbidity exponent. The default value of 1.14 is suggested
+        in [1]_ for a rural aerosol model. [unitless]
+    wavelength_variation_factor : numeric, default 0.095
+        Wavelength variation factor [unitless]
+    aerosol_asymmetry_factor : numeric, default 0.65
+        Aeorosol asymmetry factor (mean cosine of scattering angle) [unitless]
+
+    Returns
+    -------
+    spectra : dict
+        A dict of arrays of length 122. All values are spectral irradiance
```

Comment:
Except for `wavelength` would each array's shape be (len(apparent_zenith), 122)? Maybe I don't understand.